pkg/util/lock_pool.go

```diff
@@ -0,0 +1,47 @@
+// Copyright (c) 2023 VMware, Inc. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package util
+
+import (
+	"reflect"
+	"sync"
+)
+
+// LockPool is a synchronized set of maps that can be obtained with a provided
+// ID.
+type LockPool[K any, V sync.Locker] struct {
+	ool sync.Map
```

Comment:
I believe it's named as this way so that we can call `p.ool` in its methods. I will let @akutz confirm this as the code is's originally from his branch at https://github.com/akutz/vm-operator/blob/feature/generic-cache/pkg/util/lock_pool.go

Reply:
I see, it looks good for private variables.